pkg/parse/quote.go

```diff
@@ -99,26 +104,33 @@ func rtohex(r rune, w int) []byte {
 func quoteDouble(s string) string {
 	var buf bytes.Buffer
 	buf.WriteByte('"')
-	for _, r := range s {
-		if e, ok := doubleUnescape[r]; ok {
-			// Takes care of " and \ as well.
-			buf.WriteByte('\\')
-			buf.WriteRune(e)
-		} else if !unicode.IsPrint(r) {
-			buf.WriteByte('\\')
-			if r <= 0xff {
+	for len(s) > 0 {
+		r, w := utf8.DecodeRuneInString(s)
+		if r != utf8.RuneError { // the common case -- a valid UTF-8 sequence was decoded
```

Comment:
Handle the case of `r == utf8.RuneError` first, and you won't need an extra level of indentation.

Also please revert the change to the comment style; avoiding line-end comments is a conscious stylistic choice of mine.

Reply:
Agreed. In my defense it seemed sensible to handle the common case first but I agree that it makes more sense, on looking at the code again, to handle the error case first since it is very short and we can simply add a `continue` to that block and thus avoid indenting the common case.